imap_processing/lo/l0/utils/bit_decompression.py

```diff
@@ -0,0 +1,51 @@
+"""Decompression for 8 to 12, 8 to 16, and 12 to 16 bits."""
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+
+
+class Decompress(Enum):
+    """Decompression options."""
+
+    DECOMPRESS8TO12 = "8_to_12"
+    DECOMPRESS8TO16 = "8_to_16"
+    DECOMPRESS12TO16 = "12_to_16"
+
+
+def decompress_int(compressed_value, decompression):
+    """
+    Decompress a data field using a specified bit conversion.
+
+    Parameters
+    ----------
+    compressed_value : int
+        A compressed integer.
+    decompression: Decompress
+        The decompression to use.
+
+    Returns
+    -------
+    decompressed : int
+        The decompressed integer.
+    """
+    valid_decompression = [
+        Decompress.DECOMPRESS8TO12,
+        Decompress.DECOMPRESS8TO16,
+        Decompress.DECOMPRESS12TO16,
+    ]
+    if decompression not in valid_decompression:
+        raise ValueError(
+            "Invalid decompression method. Must be one of the following Enums: "
+            + "Decompress.DECOMPRESS8TO12, Decompress.DECOMPRESS8TO12, "
+            + "Decompress.DECOMPRESS8TO12"
+        )
+
+    # load the decompression table for the method specified
+    data = np.loadtxt(
+        Path(__file__).parent
+        / f"../decompression_tables/{decompression.value}_bit.csv",
+        delimiter=",",
+        skiprows=1,
+    )
```

Comment:
This disk i/o will be called a lot here, so it might help to only load the table once and store it for later use somewhere.
```python
decompression_lookup = {DecompressionEnum: np.load_txt(...) for enum in Decompress}
```

You can bring the `..` up a level to call `.parent` again.
```suggestion
    # load the decompression table for the method specified
    data = np.loadtxt(
        Path(__file__).parent.parent
        / f"decompression_tables/{decompression.value}_bit.csv",
        delimiter=",",
        skiprows=1,
    )
```

Reply:
Ok sounds good. I'll make a new function to load all the tables and call that at the top of the ScienceCounts and StarSensor files, so it only needs to be read once per packet.